storage/ndb/src/ndbapi/NdbInterpretedCode.cpp

```diff
@@ -281,6 +282,346 @@ NdbInterpretedCode::sub_reg(Uint32 RegDest,
                                RegSource2 % MaxReg));
 }
 
+int
+NdbInterpretedCode::lshift_reg(Uint32 RegDest,
+                               Uint32 RegSource1, Uint32 RegSource2)
+{
+  return add1(Interpreter::Lshift(RegDest % MaxReg, RegSource1 % MaxReg,
```

Comment:
Just wanted to note that this silent modulo MaxReg (=8) is a bit dangerous.
So when one tries to use more registers, it silently is mapped to 0 to 7, with strange results.
Max

Reply:
One comment today: this turned to be a major pain point in connection with another thing:

The new ReadFull wants the memory-offset in a register. I used a constant and got all the strangest memory errors. It didn't complain, although I tried to use register 35000 as this was translated to register 0. But that register would just hold a very low number between 0 and 10.

Reply:
I am aware that programming the interpreter is currently a tad hard, my test program is hard to read since it is all numbers, so I am thinking about possible paths to clarify it

Reply:
Those BadRegister checks of your recent commit 1e03e28e are a good improvement I think! I would have noticed this problem much earlier.

well, yes, all numbers but you can store them still in c++ variables right? Makes it a bit more readable.
In Lisp I also use register names, that is variables with readable name and then the register number as their values.

Well, for me all labels and jumps are generated now, which makes it much easier to read already, then just the linear list of instructions.